sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SchemaPruning.scala

```diff
@@ -120,7 +120,8 @@ object SchemaPruning {
   private def getRootFields(expr: Expression): Seq[RootField] = {
```

Comment:
I think `getRootFields` is used for data source v1 and v2. Why only v2 is affected?